src/UserSpaceInstrumentation/Trampoline.cpp

```diff
@@ -1113,6 +1167,13 @@ ErrorMessageOr<uint64_t> CreateTrampoline(pid_t pid, uint64_t function_address,
                                           uint64_t entry_payload_function_address,
                                           uint64_t return_trampoline_address, csh capstone_handle,
                                           absl::flat_hash_map<uint64_t, uint64_t>& relocation_map) {
+  const bool harmful_jump =
+      CheckForJumpIntoFirstFiveBytes(function_address, function, capstone_handle);
+  if (harmful_jump) {
+    return ErrorMessage(
+        "Failed to create trampoline since the function contains a jump into its own prolog.");
```

Comment:
mega nit. I wonder if "its own prolog" is the right wording, a hand written assembly function might not have a prolog at all, maybe mention a "jump into its first five bytes"?

Reply:
Agree, the prologue could be longer or shorter than five bytes.